### PR TITLE
Clear callstack and selection views

### PR DESCRIPTION
**Pull request by IrinaShkviro**
Fixed the previous PR, please have another look.

Store CallStack value in the CallStackDataView instead of
std::unique_ptr<>, remove extra FireRefreshCallbacks call in StopCapture
and clear CallStack and Selection views when clear capture or start a
new one.

Test: Take a capture, make a selection, clear capture and check the
content of the sampling, selection and callstack views; take a capture,
make a selection, start a new capture and check the content of the
sampling, selection and callstack views while capture in process.

Bug: http://b/164971132

### CODE REVIEW

File: OrbitGl/App.cpp [hunk 1]
```diff
@@ -607,14 +615,19 @@ void OrbitApp::ClearCapture() {
   AddTopDownView(*empty_sampling_profiler);
   Capture::GSamplingProfiler = empty_sampling_profiler;
 
+  if (selection_report_) {
+    auto empty_selection_profiler = std::make_shared<SamplingProfiler>(Capture::GTargetProcess);
```
Comment (florian-kuebler) [inline OrbitGl/App.cpp:619]: If #1077 will be in before this change, it will silently lead to compilation error, as GTargetProcess will be gone. Otherwise, I'll take care of updating #1077.

**Review by florian-kuebler — APPROVED** [review]
LGTM. Thanks!

File: OrbitGl/App.cpp (hunk 1 above)
Comment (IrinaShkviro) [reply OrbitGl/App.cpp:619]: I can wait for your PR and rebase then

File: OrbitGl/App.cpp [hunk 2]
```diff
@@ -383,6 +385,10 @@ void OrbitApp::AddSamplingReport(std::shared_ptr<SamplingProfiler> sampling_prof
     sampling_reports_callback_(callstack_data_view, report);
   }
 
+  // clear old sampling report
```
Comment (dimitry-) [inline OrbitGl/App.cpp:388]: why is this needed? maybe add a comment explaining?

Comment (IrinaShkviro) [reply OrbitGl/App.cpp:388]: The main reason for this is clearing CallStackDataView (the field `CallStackDataView* callstack_data_view_;` in the SamplingReport). Should I write more detailed comment here?
